### PR TITLE
feat(pull_reqeust): :sparkles: When pull request is closed add `closed` label and remove `ready for review` or `approved` label

### LINKED ISSUE #27: :sparkles: When pull request is closed add `closed` label and remove `ready for review` or `approved` label
### Description

If we close the pull request with `ready for review` label it stays there
so when we close the pull request `ready for review` should be removed and closed label should
be added

### Anything else?

image

---

image

### PR DESCRIPTION
**Pull request by aayushchugh**
# Description

When pull request is closed add `closed` label and remove `ready for review` or `approved` label
Closes #27 

# Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (postman)

# Screenshots

image

### CODE REVIEW

File: automation/addLabelsOnPullRequest.automation.js [hunk 1]
```diff
@@ -102,63 +101,63 @@ exports.pullRequestWIPLabelAutomation = async context => {
...
 
 	if (
-		(title.includes('WIP') ||
-			title.includes('Work In Progress') ||
-			title.includes('work in progress') ||
-			title.includes(':construction:')) &&
+		(title.includes("WIP") ||
+			title.includes("Work In Progress") ||
+			title.includes("work in progress") ||
+			title.includes(":construction:")) &&
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:114]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
      title.includes(":construction:")) &&
```

File: index.js [hunk 2]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
+	app.on("pull_request.opened", addReadyForReviewLabel);
 
-	app.on('pull_request_review.submitted', addApprovedLabel);
-	app.on('pull_request_review.submitted', changesRequestLabel);
+	app.on("pull_request_review.submitted", addApprovedLabel);
+	app.on("pull_request_review.submitted", changesRequestLabel);
 
-	app.on('pull_request.closed', addMergedLabel);
+	app.on("pull_request.closed", addMergedLabel);
+	app.on("pull_request.closed", async context => {
```
Comment (aayushchugh) [inline index.js:38]: Codacy has a fix for the issue: [Replace `↹app.on("pull_request.closed",·async·context` with `··app.on("pull_request.closed",·async·(context)`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
  app.on("pull_request.closed", async (context) => {
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 3]
```diff
@@ -102,63 +101,63 @@ exports.pullRequestWIPLabelAutomation = async context => {
...
-		label => label.name === ':mag: Ready for Review'
+		label => label.name === ":mag: Ready for Review"
 	);
 
 	if (
-		(title.includes('WIP') ||
-			title.includes('Work In Progress') ||
-			title.includes('work in progress') ||
-			title.includes(':construction:')) &&
+		(title.includes("WIP") ||
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:111]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
    (title.includes("WIP") ||
```

File: index.js [hunk 4]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
 
 	/* --------------------------------- ANCHOR Issue commands --------------------------------- */
-	app.on('issue_comment.created', async context => {
+	app.on("issue_comment.created", async context => {
 		const { body } = context.payload.comment;
 		const { command, args } = getCommandAndArgs(body);
 		const commentId = context.payload.comment.id;
 
-		if (command[0] === '/' && !context.isBot) {
+		if (command[0] === "/" && !context.isBot) {
```
Comment (aayushchugh) [inline index.js:86]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
    if (command[0] === "/" && !context.isBot) {
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 5]
```diff
@@ -102,63 +101,63 @@ exports.pullRequestWIPLabelAutomation = async context => {
...
-			removeLabel([':mag: Ready for Review'], context);
+			removeLabel([":mag: Ready for Review"], context);
 		}
 
 		return;
 	}
 
 	if (foundReadyForReviewLabel.length > 0 && foundWIPLabel.length > 0) {
-		return removeLabel([':mag: Ready for Review'], context);
+		return removeLabel([":mag: Ready for Review"], context);
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:160]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
    return removeLabel([":mag: Ready for Review"], context);
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 6]
```diff
@@ -170,36 +169,36 @@ exports.changesRequestLabel = async context => {
...
-		label => label.name === ':white_check_mark: Approved'
+		label => label.name === ":white_check_mark: Approved"
 	);
 
 	if (
 		changesRequestedReviews.length > 0 &&
 		foundChangedRequestedLabel.length === 0
 	) {
-		addLabel([':warning: Changes requested'], context, 'AA2626');
+		addLabel([":warning: Changes requested"], context, "AA2626");
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:191]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
    addLabel([":warning: Changes requested"], context, "AA2626");
```

File: index.js [hunk 7]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
 				break;
 
 			default:
-				if (command[0] === '/') {
+				if (command[0] === "/") {
 					if (!context.isBot) {
 						const params = context.issue({
 							comment_id: commentId,
-							content: '-1',
+							content: "-1",
```
Comment (aayushchugh) [inline index.js:117]: Codacy has a fix for the issue: [Replace `↹↹↹↹↹↹↹` with `··············`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
              content: "-1",
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 8]
```diff
@@ -102,63 +101,63 @@ exports.pullRequestWIPLabelAutomation = async context => {
...
 		return;
 	}
 
 	if (
 		foundWIPLabel.length > 0 &&
-		!title.includes('WIP') &&
-		!title.includes('Work In Progress') &&
-		!title.includes('work in progress') &&
-		!title.includes(':construction:')
+		!title.includes("WIP") &&
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:128]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
    !title.includes("WIP") &&
```

File: index.js [hunk 9]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
-			case '/approve':
+			case "/approve":
 				approveCommand(context);
 				break;
-			case '/merge':
+			case "/merge":
 				mergeCommand(context);
 				break;
-			case '/WIP':
+			case "/WIP":
```
Comment (aayushchugh) [inline index.js:108]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
      case "/WIP":
```

File: index.js [hunk 10]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
 		const { body } = context.payload.comment;
 		const { command, args } = getCommandAndArgs(body);
 		const commentId = context.payload.comment.id;
 
-		if (command[0] === '/' && !context.isBot) {
+		if (command[0] === "/" && !context.isBot) {
 			const params = context.issue({
 				comment_id: commentId,
-				content: '+1',
+				content: "+1",
```
Comment (aayushchugh) [inline index.js:89]: Codacy has a fix for the issue: [Replace `↹↹↹↹` with `········`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
        content: "+1",
```

File: index.js [hunk 11]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
 				mergeCommand(context);
 				break;
-			case '/WIP':
+			case "/WIP":
 				WIPCommand(context);
 				break;
 
 			default:
-				if (command[0] === '/') {
+				if (command[0] === "/") {
```
Comment (aayushchugh) [inline index.js:113]: Codacy has a fix for the issue: [Replace `↹↹↹↹` with `········`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
        if (command[0] === "/") {
```

File: index.js [hunk 12]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
-				content: '+1',
+				content: "+1",
 			});
 
 			context.octokit.reactions.createForIssueComment(params);
 		}
 
 		switch (command) {
-			case '/label':
+			case "/label":
```
Comment (aayushchugh) [inline index.js:96]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
      case "/label":
```

File: index.js [hunk 13]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
+		["pull_request.edited", "pull_request.labeled"],
 		pullRequestWIPLabelAutomation
 	);
 
-	app.on('issues.closed', addLabelToIssueOnClose);
+	app.on("issues.closed", addLabelToIssueOnClose);
 
 	/* --------------------------------- ANCHOR Issue commands --------------------------------- */
-	app.on('issue_comment.created', async context => {
+	app.on("issue_comment.created", async context => {
```
Comment (aayushchugh) [inline index.js:81]: Codacy has a fix for the issue: [Replace `↹app.on("issue_comment.created",·async·context` with `··app.on("issue_comment.created",·async·(context)`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
  app.on("issue_comment.created", async (context) => {
```

File: index.js [hunk 14]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
-			case '/label':
+			case "/label":
 				labelCommand(context, args);
 				break;
-			case '/close':
+			case "/close":
 				closeCommand(context);
 				break;
-			case '/approve':
+			case "/approve":
```
Comment (aayushchugh) [inline index.js:102]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
      case "/approve":
```

File: index.js [hunk 15]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
 			context.octokit.reactions.createForIssueComment(params);
 		}
 
 		switch (command) {
-			case '/label':
+			case "/label":
 				labelCommand(context, args);
 				break;
-			case '/close':
+			case "/close":
```
Comment (aayushchugh) [inline index.js:99]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
      case "/close":
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 16]
```diff
@@ -65,31 +65,30 @@ exports.addMergedLabel = async context => {
...
+			removeLabel([":construction: WIP"], context);
 		}
-		console.log(title);
 
 		if (
-			title.includes('WIP') ||
-			title.includes('Work In Progress') ||
-			title.includes('work in progress') ||
-			title.includes(':construction:')
+			title.includes("WIP") ||
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:85]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
      title.includes("WIP") ||
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 17]
```diff
@@ -102,63 +101,63 @@ exports.pullRequestWIPLabelAutomation = async context => {
...
-			title.includes('work in progress') ||
-			title.includes(':construction:')) &&
+		(title.includes("WIP") ||
+			title.includes("Work In Progress") ||
+			title.includes("work in progress") ||
+			title.includes(":construction:")) &&
 		foundWIPLabel.length === 0
 	) {
-		addLabel([':construction: WIP'], context);
+		addLabel([":construction: WIP"], context);
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:117]: Codacy has a fix for the issue: [Replace `↹↹` with `····`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
    addLabel([":construction: WIP"], context);
```

File: index.js [hunk 18]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
 				closeCommand(context);
 				break;
-			case '/approve':
+			case "/approve":
 				approveCommand(context);
 				break;
-			case '/merge':
+			case "/merge":
```
Comment (aayushchugh) [inline index.js:105]: Codacy has a fix for the issue: [Replace `↹↹↹` with `······`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
      case "/merge":
```

File: automation/addLabelsOnPullRequest.automation.js [hunk 19]
```diff
@@ -65,31 +65,30 @@ exports.addMergedLabel = async context => {
...
-			title.includes('work in progress') ||
-			title.includes(':construction:')
+			title.includes("WIP") ||
+			title.includes("Work In Progress") ||
+			title.includes("work in progress") ||
+			title.includes(":construction:")
 		) {
 			const params = context.pullRequest({
-				title: `${title.replace(':construction:', '')}`,
+				title: `${title.replace(":construction:", "")}`,
```
Comment (aayushchugh) [inline automation/addLabelsOnPullRequest.automation.js:91]: Codacy has a fix for the issue: [Replace `↹↹↹↹` with `········`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
        title: `${title.replace(":construction:", "")}`,
```

File: index.js [hunk 20]
```diff
@@ -29,58 +29,92 @@ const availableCommandsMessage = `Available commands are:-
...
+	});
 
 	app.on(
-		['pull_request.edited', 'pull_request.labeled'],
+		["pull_request.edited", "pull_request.labeled"],
 		pullRequestWIPLabelAutomation
 	);
 
-	app.on('issues.closed', addLabelToIssueOnClose);
+	app.on("issues.closed", addLabelToIssueOnClose);
```
Comment (aayushchugh) [inline index.js:78]: Codacy has a fix for the issue: [Replace `↹` with `··`](https://app.codacy.com/gh/aayushchugh/bot/pullRequest?prid=10060881)
```suggestion
  app.on("issues.closed", addLabelToIssueOnClose);
```

**Review by Arpitchugh — APPROVED** [review]

**Pull request merged**